packages/amplify-graphql-auth-transformer/src/resolvers/query.ts

```diff
@@ -266,13 +268,32 @@ const generateAuthFilter = (roles: Array<RoleDefinition>, fields: ReadonlyArray<
   roles.forEach((role, idx) => {
     const entityIsList = fieldIsList(fields, role.entity);
     if (role.strategy === 'owner') {
+      const claims = role.claim!.split(':');
+      const hasMultiClaims = claims.length > 1 && role.claim !== 'cognito:username';
       const ownerCondition = entityIsList ? 'contains' : 'eq';
+
+      if (hasMultiClaims && !entityIsList) {
+        claims.forEach((claim, secIdx) => {
+          authCollectionExp.push(
+            ...[
+              set(ref(`role${idx}_${secIdx}`), getOwnerClaim(claim)),
+              set(ref(`ownerPrefix${idx}_${secIdx}`), str(`$role${idx}_${secIdx}:`)),
+              iff(
+                notEquals(ref(`role${idx}_${secIdx}`), str(NONE_VALUE)),
+                qref(methodCall(ref('authFilter.add'), raw(`{"${role.entity}": { "beginsWith": $ownerPrefix${idx}_${secIdx} }}`))),
+              ),
+            ],
+          );
+        });
+      }
+      const secIdx = claims.length;
+
       authCollectionExp.push(
         ...[
-          set(ref(`role${idx}`), getOwnerClaim(role.claim!)),
+          set(ref(`role${idx}_${secIdx}`), getOwnerClaim(role.claim!)),
           iff(
-            notEquals(ref(`role${idx}`), str(NONE_VALUE)),
-            qref(methodCall(ref('authFilter.add'), raw(`{"${role.entity}": { "${ownerCondition}": $role${idx} }}`))),
+            notEquals(ref(`role${idx}_${secIdx}`), str(NONE_VALUE)),
+            qref(methodCall(ref('authFilter.add'), raw(`{"${role.entity}": { "${ownerCondition}": $role${idx}_${secIdx} }}`))),
           ),
         ],
       );
```

Comment:
should this be in an 'else' block from the if online 275?

Reply:
This is the "backwards compatible" part - it adds both a filter to look for the "sub" in the owner field, and it will also use "eq" or "contains" (for list) to do an equality check.

Reply:
Also want to note here that compatibility with lists comes with mutation because it has to build the `<sub>:<username>` value which is in https://github.com/aws-amplify/amplify-cli/pull/10196